go/private/mode.bzl

```diff
@@ -51,24 +29,61 @@ def mode_string(mode):
     result.append("msan")
   if mode.pure:
     result.append("pure")
+  if mode.debug:
+    result.append("debug")
+  if mode.strip:
+    result.append("stripped")
   if not result or not mode.link == LINKMODE_NORMAL:
     result.append(mode.link)
   return "_".join(result)
 
-NORMAL_MODE = DEFAULT_MODE
-RACE_MODE = mode(base = DEFAULT_MODE, race=True)
-STATIC_MODE = mode(base = DEFAULT_MODE, static=True)
-
-common_modes = (NORMAL_MODE, RACE_MODE, STATIC_MODE)
+def _ternary_mode(values):
```

Comment:
Several comments and questions:

* Is there any case where this would accept anything other than bools and the strings "on", "off", and "auto"? If not, let's not accept "true", "false", "yes", and "no".
* Consider renaming to just `_ternary` or `_tristate`. The current name is a little confusing, since this function doesn't operate directly on modes.
* Consider accepting `*args`.
* fail if the loop exits.

Reply:
true and false were because the bools were getting stringified, but I fixed it a while ago, yes and no were because I kept changing my mind :)